flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/AbstractCollectingResultPartitionWriter.java

```diff
@@ -70,10 +70,11 @@ public BufferBuilder getBufferBuilder() throws IOException, InterruptedException
 	}
 
 	@Override
-	public synchronized void addBufferConsumer(BufferConsumer bufferConsumer, int targetChannel) throws IOException {
+	public synchronized boolean addBufferConsumer(BufferConsumer bufferConsumer, int targetChannel) throws IOException {
 		checkState(targetChannel < getNumberOfSubpartitions());
-		bufferConsumers.add(bufferConsumer);
+		boolean success = bufferConsumers.add(bufferConsumer);
```

Comment:
why do we return this value as the method's return value?

Reply:
I think it has two possibilities here.  One is return true, and the other is throwing an exception. I could return true directly. WDYT?